tests/integration_tests/performance/test_snapshot_perf.py

```diff
@@ -0,0 +1,250 @@
+# SPDX-License-Identifier: Apache-2.0
+"""Basic tests scenarios for snapshot save/restore."""
+
+import logging
+import os
+import platform
+import pytest
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.matrix import TestMatrix, TestContext
+from framework.builder import MicrovmBuilder, SnapshotBuilder, SnapshotType
+import host_tools.network as net_tools  # pylint: disable=import-error
+import host_tools.logging as log_tools
+
+
+# How many latencies do we sample per test.
+SAMPLE_COUNT = 3
+
+# Latencies in milliseconds.
+CREATE_LATENCIES = {
+    '2vcpu_256mb.json': {
+        'FULL':  250,
+        'DIFF':  0,
+    },
+    '2vcpu_512mb.json': {
+        'FULL':  500,
+        'DIFF':  0,
+    }
+}
+
+# The latencies are pretty high during integration tests and
+# this is tracked here:
+# https://github.com/firecracker-microvm/firecracker/issues/2027
+# TODO: Update the table after fix. Target is < 5ms.
+LOAD_LATENCIES = {
+    '2vcpu_256mb.json': 35,
+    '2vcpu_512mb.json': 35,
+}
+
+
+def _test_snapshot_create_latency(context):
+    logger = context.custom['logger']
+    vm_builder = context.custom['builder']
+    snapshot_type = context.custom['snapshot_type']
+    enable_diff_snapshots = snapshot_type == SnapshotType.DIFF
+
+    logger.info("""Measuring snapshot create({}) latency for microvm: \"{}\",
+kernel {}, disk {} """.format(snapshot_type,
+                              context.microvm.name(),
+                              context.kernel.name(),
+                              context.disk.name()))
+
+    # Create a rw copy artifact.
+    rw_disk = context.disk.copy()
+    # Get ssh key from read-only artifact.
+    ssh_key = context.disk.ssh_key()
+
+    # Measure a burst of snapshot create calls.
+    for i in range(SAMPLE_COUNT):
+        # Create a fresh microvm from aftifacts.
+        vm = vm_builder.build(kernel=context.kernel,
+                              disks=[rw_disk],
+                              ssh_key=ssh_key,
+                              config=context.microvm,
+                              enable_diff_snapshots=enable_diff_snapshots)
+
+        # Configure metrics system.
+        metrics_fifo_path = os.path.join(vm.path, 'metrics_fifo')
+        metrics_fifo = log_tools.Fifo(metrics_fifo_path)
+
+        response = vm.metrics.put(
+            metrics_path=vm.create_jailed_resource(metrics_fifo.path)
+        )
+        assert vm.api_session.is_status_no_content(response.status_code)
+
+        vm.start()
+
+        # Create a snapshot builder from a microvm.
+        snapshot_builder = SnapshotBuilder(vm)
+        snapshot_builder.create([rw_disk],
+                                ssh_key,
+                                snapshot_type)
+        metrics = vm.flush_metrics(metrics_fifo)
+
+        if snapshot_type == SnapshotType.FULL:
+            value = metrics['latencies_us']['full_create_snapshot'] / 1000
```

Comment:
`1000` should be a constant, such as:
`USEC_IN_MSEC = 1000`

Reply:
done